.github/workflows/pr.yaml

```diff
@@ -150,7 +150,7 @@ jobs:
   unit-tests-cudf-pandas:
     needs: wheel-build-cudf
     secrets: inherit
-    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.04
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@python-3.11
     with:
       matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10" and (.CUDA_VER == "11.8.0" or .CUDA_VER == "12.2.2")))
```

Comment:
For all our `matrix_filter` entries, we'll want to update 3.10 to 3.11. This keeps those jobs running on the "latest" matrix configuration. Let's grep for those.

Reply:
Oh. Sorry. This PR isn't for Python 3.11, it's for the refactored test matrix.

Reply:
I'm sorry, I made this really confusing.

I *did* originally open this to test just the matrix-refactoring (https://github.com/rapidsai/shared-workflows/pull/177).

But then since I already had it up, I used it to also test the Python 3.11 additions from https://github.com/rapidsai/shared-workflows/pull/176. And by "test", I just narrowly meant "check that all the jobs that I expect to get triggered are triggered".

I'll update this with https://github.com/rapidsai/rapids-reviser/pull/11 to make it a better test of what would happen if we merged https://github.com/rapidsai/shared-workflows/pull/176. e.g. update this filter, change `dependencies.yaml`, etc.

So for now... @bdice could you look at this: https://github.com/bdice/rapids-reviser/pull/1